Content.Shared/GameObjects/Components/Buckle/SharedBuckleComponent.cs

```diff
@@ -44,19 +67,23 @@ bool IDraggable.CanDrop(CanDropEventArgs args)
         public bool Drop(DragDropEventArgs args)
         {
             return TryBuckle(args.User, args.Target);
-        }
+        }  
     }
 
     [Serializable, NetSerializable]
     public sealed class BuckleComponentState : ComponentState
     {
-        public BuckleComponentState(bool buckled, int? drawDepth) : base(ContentNetIDs.BUCKLE)
+        public BuckleComponentState(bool buckled, int? drawDepth, EntityUid? entityBuckledTo, bool dontCollide) : base(ContentNetIDs.BUCKLE)
         {
             Buckled = buckled;
             DrawDepth = drawDepth;
+            EntityBuckledTo = entityBuckledTo;
+            DontCollide = dontCollide;
         }
 
         public bool Buckled { get; }
```

Comment:
As you are passing the entity buckled to through the component state now, remove the `bool buckled` argument from the constructor and change the property to be `public bool Buckled => EntityBuckledTo != null;`

Reply:
I cannot make this change for the same reason as above, EntityBuckledTo is not the actual Entity but the last Entity the player was buckled to.

Reply:
Resolved, see https://github.com/space-wizards/space-station-14/pull/2694#discussion_r539560661